Simulator/simulator.rb

```diff
@@ -0,0 +1,149 @@
+require_relative '../Account/account'
+require_relative '../User/user'
+class Simulator
+
+  def initialize
+    @user_controller = UserController.new
+    @account_controller = AccountController.new(@user_controller)
+
+  end
+
+  def login
+    puts "ATM Simulator!"
+    puts "Please login to your account"
+    puts "Please enter your name:"
+    name = gets.chomp
+    puts "Please enter your password:"
+    password = gets.chomp
+    @user_controller.login(name, password)
+  end
+  def start(user)
+    if user.type == "admin"
+      loop do
+        puts "Please select an option:"
+        puts "1. Create User"
+        puts "2. Edit User"
+        puts "3. Delete User"
+        puts "4. Create User account"
+        puts "5. Exit"
+        choice = gets.chomp.to_i
+
+        case choice
+        when 1
+          puts create_user
+        when 2
+          puts edit_user
+        when 3
+          puts delete_user
+        when 4
+          puts create_user_account
+        when 5
+          stop
+          break
+        else
+          puts "Invalid choice. Please try again."
+        end
+      end
+      puts "Thank you for using ATM!"
+
+    else
+      loop do
+        puts "Please select an option:"
+        puts "1. Make a transaction"
+        puts "2. Exit"
+        choice = gets.chomp.to_i
+
+        case choice
+        when 1
+          puts make_a_transaction(user)
+        when 2
+          stop
+          break
+        else
+          puts "Invalid choice. Please try again."
+        end
+      end
+      puts "Thank you for using ATM!"
+    end
+
+  end
+
+  def create_user
+    puts "Please enter the user's name:"
+    name = gets.chomp
+
+    puts "Please enter the user's address:"
+    address = gets.chomp
+
+    puts "Please enter the user's national id:"
+    national_id = gets.chomp
+
+    puts "Please enter the user's password:"
+    password = gets.chomp
```

Comment:
Although this gets the job done but you could use a user hash here instead of individual arguments.